### PR TITLE
fix(discover): Display error message on failed requests

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx [hunk 1]
```diff
@@ -147,18 +147,24 @@ export default class OrganizationDiscover extends React.Component {
...
-        browserHistory.push({
-          pathname: `/organizations/${organization.slug}/discover/`,
-          search: getQueryStringFromQuery(queryBuilder.getInternal()),
-        });
-      }
-
-      this.setState({data, isFetchingQuery: false});
-    });
+    resultManager
+      .fetchAll()
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx]: Hmm, not sure why this got autoformatted

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx]: Prob from adding another chain

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx [hunk 2]
```diff
@@ -147,18 +147,24 @@ export default class OrganizationDiscover extends React.Component {
...
+        if (shouldRedirect) {
+          browserHistory.push({
+            pathname: `/organizations/${organization.slug}/discover/`,
+            search: getQueryStringFromQuery(queryBuilder.getInternal()),
+          });
+        }
+
+        this.setState({data, isFetchingQuery: false});
+      })
+      .catch(err => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx]: What's `err` here? Is it going to be a response from the API?

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx]: Yes, either that or it's failed one of the client side validations

**Pull request merged**